### ISSUE TITLE
clamav pod does not run on minikube when using the provided chart for helm

### ISSUE THREAD

**Issue by csegarra** [nextcloud/all-in-one#3422]
### Steps to reproduce
1. Install docker following [the oficial guide](https://docs.docker.com/engine/install/debian/)
2. Install minikube following [the oficial guide](https://minikube.sigs.k8s.io/docs/start/)
3. Install helm following [the oficial guide](https://helm.sh/docs/intro/install/)
4. Start minikube
5. Generate a `values.yaml` file for helm
6. Install Nextcloud AIO using helm

```
helm install aio nextcloud-aio/nextcloud-aio-helm-chart --version 7.2.1 -f ~/values.yaml
```

### Expected behavior

All pods up and running.

Something like this:

```
$ minikube kubectl -- get pods --namespace aio

NAME                                            READY   STATUS             RESTARTS     AGE
nextcloud-aio-apache-7d55c8d64f-m48w9           1/1     Running            0            74s
nextcloud-aio-clamav-f6c94b65c-x9sjx            1/1     Running            0            75s
nextcloud-aio-collabora-84d57cb6ff-m5dn9        1/1     Running            0            75s
nextcloud-aio-database-844fddfbc9-lz768         1/1     Running            0            74s
nextcloud-aio-fulltextsearch-6db7474d64-zv68p   1/1     Running            0            74s
nextcloud-aio-imaginary-7879c65d6b-578d6        1/1     Running            0            74s
nextcloud-aio-nextcloud-5d955bfd56-2qr9k        1/1     Running            0            75s
nextcloud-aio-notify-push-dc984b59-4vzhx        1/1     Running            0            75s
nextcloud-aio-onlyoffice-fd8fb84cf-g897v        1/1     Running            0            74s
nextcloud-aio-redis-785d678d4f-7qll4            1/1     Running            0            74s
nextcloud-aio-talk-58fcb5dcb9-9hdb4             1/1     Running            0            74s
nextcloud-aio-talk-recording-86754d79fb-jx2cp   1/1     Running            0            75s
```

### Actual behavior

The `nextcloud-aio-clamav` pod is failing to start and it's status stays as `CrashLoopBackOff`.

```
$ minikube kubectl -- get pods --namespace aio
NAME                                            READY   STATUS             RESTARTS     AGE
nextcloud-aio-apache-7d55c8d64f-m48w9           1/1     Running            0            74s
nextcloud-aio-clamav-f6c94b65c-x9sjx            0/1     CrashLoopBackOff   3 (6s ago)   75s
nextcloud-aio-collabora-84d57cb6ff-m5dn9        1/1     Running            0            75s
nextcloud-aio-database-844fddfbc9-lz768         1/1     Running            0            74s
nextcloud-aio-fulltextsearch-6db7474d64-zv68p   1/1     Running            0            74s
nextcloud-aio-imaginary-7879c65d6b-578d6        0/1     Pending            0            74s
nextcloud-aio-nextcloud-5d955bfd56-2qr9k        1/1     Running            0            75s
nextcloud-aio-notify-push-dc984b59-4vzhx        1/1     Running            0            75s
nextcloud-aio-onlyoffice-fd8fb84cf-g897v        1/1     Running            0            74s
nextcloud-aio-redis-785d678d4f-7qll4            1/1     Running            0            74s
nextcloud-aio-talk-58fcb5dcb9-9hdb4             1/1     Running            0            74s
nextcloud-aio-talk-recording-86754d79fb-jx2cp   1/1     Running            0            75s
```

The logs of the pod reports a permissions problem when executing a *chown* at the */var/lib/clamav* mountpoint.

```
$ minikube kubectl -- --namespace aio logs --previous nextcloud-aio-clamav-f6c94b65c-x9sjx
Defaulted container "nextcloud-aio-clamav" out of: nextcloud-aio-clamav, init-volumes (init)
chown: /var/lib/clamav: Operation not permitted
chown: /var/lib/clamav: Operation not permitted
```

The error is on the `nextcloud-aio-clamav` container, not the init volume of the pod.

```
$ minikube kubectl -- --namespace aio-7 describe pod nextcloud-aio-clamav-f6c94b65c-x9sjx
[...]
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  19m                   default-scheduler  Successfully assigned aio-7/nextcloud-aio-clamav-f6c94b65c-x9sjx to minikube
  Normal   Pulling    19m                   kubelet            Pulling image "alpine"
  Normal   Pulled     19m                   kubelet            Successfully pulled image "alpine" in 2.686665552s (5.141532873s including waiting)
  Normal   Created    19m                   kubelet            Created container init-volumes
  Normal   Started    19m                   kubelet            Started container init-volumes
  Normal   Created    18m (x4 over 19m)     kubelet            Created container nextcloud-aio-clamav
  Normal   Started    18m (x4 over 18m)     kubelet            Started container nextcloud-aio-clamav
  Normal   Pulled     17m (x5 over 19m)     kubelet            Container image "nextcloud/aio-clamav:20230916_091439-latest" already present on machine
  Warning  BackOff    3m59s (x71 over 18m)  kubelet            Back-off restarting failed container nextcloud-aio-clamav in pod nextcloud-aio-clamav-f6c94b65c-x9sjx_aio-7(a17248d4-f628-4252-84ae-eeeae7668764)
```

### Host OS

Debian 12.1

#### Nextcloud AIO chart version

I have tried versions `7.0.0`, `7.1.1` and `7.2.1` of the Nextcloud AIO helm chart with the same results.

#### Other valuable info

I have attached the config file used with helm and the output od the `logs` and `describe` kubectls commands.

[nextcloud-aio-clamav-describe.txt](https://github.com/nextcloud/all-in-one/files/12739588/nextcloud-aio-clamav-describe.txt)
[nextcloud-aio-clamav-logs.txt](https://github.com/nextcloud/all-in-one/files/12739589/nextcloud-aio-clamav-logs.txt)
[values.txt](https://github.com/nextcloud/all-in-one/files/12739591/values.txt)

**Comment by csegarra** [comment]
I have build a custom container for clamav after removing the `USER clamav` line and then the pod runs ok.

```
cat Containers/clamav/Dockerfile
# Probably from this file: https://github.com/Cisco-Talos/clamav-docker/blob/main/clamav/1.1/alpine/Dockerfile
FROM clamav/clamav:1.2.0-1

COPY clamav.conf /tmp/clamav.conf

RUN set -ex; \
    apk add --no-cache tzdata; \
    cat /tmp/clamav.conf | tee -a /etc/clamav/clamd.conf; \
    rm /tmp/clamav.conf; \
    mkdir -p /var/run/clamav /run/lock; \
    chown -R clamav:clamav /var/run/clamav /run/clamav /var/log/clamav /var/lock /run/lock; \
    chmod 777 -R /var/run/clamav /run/clamav /var/log/clamav /var/lock /run/lock /tmp

VOLUME /var/lib/clamav

LABEL com.centurylinklabs.watchtower.enable="false"
```

I think that running clamav as `root` it's not ok, but this test may give a hint on where to start looking.

**Comment by szaimen** [comment]
Hi, which docker version is backing minikube?

**Comment by csegarra** [comment]
Hi @szaimen .

This are the versions I'm using:

* minikube v1.31.2 - commit: fd7ecd9c4599bef9f04c0986c4a0187f98a4396e
* helm v3.12.3 - commit: 3a31588ad33fe3b89af5a2a54ee1d25bfe6eaa5e
* docker 24.0.5, build ced0996

Thanks!

**Comment by szaimen** [comment]
Oh, this doesnt look good. Apparently the behaviour is then different with minikube compared to docker because when mounting a tmpfs in a specific directly it will be writeable for all users within docker. Not sure how to change the behaviour on minikube. Do you have some docs on this maybe?

**Comment by szaimen** [comment]
I think it is actually this bug: https://github.com/kubernetes/kubernetes/issues/48912

**Comment by szaimen** [comment]
So probably we will need to get rid of readOnlyRootFilesystem and tmpfs for the helm chart again because kubernetes has not fixed this yet.

**Comment by szaimen** [comment]
Hi, this is now fixed with v7.4.1. Testing and feedback is welcome!
